app/src/main/java/com/crossbowffs/nekosms/xposed/SmsHandlerHook.java

```diff
@@ -296,6 +296,19 @@ private void hookDispatchIntent23(XC_LoadPackage.LoadPackageParam lpparam) {
             /*           user */ UserHandle.class,
             new DispatchIntentHook(4));
     }
+    
+    private void hookDispatchIntent29(XC_LoadPackage.LoadPackageParam lpparam) {
+        Xlog.i("Hooking dispatchIntent() for Android v29+");
+        XposedHelpers.findAndHookMethod(SMS_HANDLER_CLASS, lpparam.classLoader, "dispatchIntent",
+                /*         intent */ Intent.class,
+                /*     permission */ String.class,
+                /*          appOp */ int.class,
```

Comment:
This should be `String` (I think? Unless you tested this and it worked?)

Reply:
I tested it here and it works

Reply:
Hm then you must have gotten an Android build based off the code between [this commit](https://android.googlesource.com/platform/frameworks/opt/telephony/+/40d7f1612f837ff233015bdda96ccb85b1e002ea%5E%21/src/java/com/android/internal/telephony/InboundSmsHandler.java) and [this commit](https://android.googlesource.com/platform/frameworks/opt/telephony/+/6939834098aaf8126da4832453ebf64a85a898a6%5E%21/src/java/com/android/internal/telephony/InboundSmsHandler.java). What ROM you are using? This does not seem to be the version in the `android10-release` branch, and others have reported that the original code works on Android Q as-is, so this will probably break it for them.

Reply:
havoc3.1

Reply:
Huh, what do you know, they actually changed this in the middle of a version. That hasn't happened before. Okay, we need to try both versions then, that is:

```
try {
    hookDispatchIntent29(lpparam);
} catch (NoSuchMethodError) {
    hookDispatchIntent23(lpparam);
}
```

We might also need to make this more generic in the future, but this should suffice for now.

Reply:
Now you can compile this version, because this method was told by others, and he found that several roms are like this, if there is time, you can do it again.

Reply:
Not sure what you mean, at the very least we need to try both modes, right?